### PR TITLE
Add address validation and matching

### PR DESCRIPTION
**Pull request by DrLuke**
Added address module with following features:
* Address validation
* Address matching

The validation is differentiates between method and message addresses, as defined by the OSC specification. Address matching uses either direct comparison or regex for addresses with wildcards.

I've tried to cover all kinds of cases with tests but I'm sure I've missed something.

👋 This is my first time contributing to a rust project, so feedback ony my code is very welcome!

### CODE REVIEW

**Comment by klingtnet** [comment]
I'm glad to hear that `rosc` will be your first Rust 🦀 contribution!  I try to review your changes as soon as possible but expect it to take a couple of days.

> The validation is differentiates between method and message addresses, as defined by the OSC specification

Can you link to the specification section where method and addresses are defined so I can verify the implementation?

**Comment by DrLuke** [comment]
> > The validation is differentiates between method and message addresses, as defined by the OSC specification
> 
> Can you link to the specification section where method and addresses are defined so I can verify the implementation?

Sure! [Here](http://opensoundcontrol.org/spec-1_0.html) what I call a *method address* is explained under **OSC Address Spaces and OSC Addresses** and a *message address* is right below in **OSC Message Dispatching and Pattern Matching**. (Sorry page doesn't have chapter anchors :disappointed: )

What surprised me is that method addresses can't contain any wildcars, only the address in a message can. I think I've always seen this implemented the other way :smile:

**Review by klingtnet — CHANGES_REQUESTED** [review]

**Comment by klingtnet** [comment]
Sorry, the GitHub UI confused me again.  I already added the review suggestions four days ago but forgot to press the _request changes_ button.

File: Cargo.toml [hunk 1]
```diff
@@ -8,10 +8,12 @@ documentation = "https://docs.rs/rosc"
 repository = "https://github.com/klingtnet/rosc"
 license = "MIT/Apache-2.0"
 readme = "README.md"
+edition = "2021"
```
[the remark this answers is not in the document]
Comment (DrLuke) [reply Cargo.toml:11]: I was wondering about that when I read that in a SO answer but then forgot about it 😅

**Comment by klingtnet** [comment]
Waits on https://github.com/DrLuke/rosc/pull/1 to be merged.

**Comment by klingtnet** [comment]
Hey 🖖🏻 

Just wanted to let you know that I'm still alive and will merge this PR soon, hopefully Sunday.  Sorry for the delay.

**Review by klingtnet — APPROVED** [review]

**Comment by klingtnet** [comment]
Finally, it's time for a merge.  This will be included with the `0.6.0` I'm about to release.
@DrLuke  Thanks again!

**Pull request merged**